src/runtime/PromiseObject.cpp

```diff
@@ -82,6 +82,14 @@ PromiseReaction::Capability PromiseObject::createResolvingFunctions(ExecutionSta
     return PromiseReaction::Capability(this, resolve, reject);
 }
 
+PromiseReaction::Capability PromiseObject::newPromiseCapability(ExecutionState& state, const Value& constructor, const Value& parentPromise)
+{
+    if (!constructor.isConstructor()) {
+        ErrorObject::throwBuiltinError(state, ErrorCode::TypeError, ErrorObject::Messages::Not_Constructor);
+    }
```

Comment:
What about using the existing `PromiseObject::newPromiseCapability(ExecutionState& state, Object* constructor, const Value& parentPromise)` function instead?
This function is invoked by only one function, and `constructor` check can be done on the caller side.

Reply:
Good point.